CMakeLists.txt

```diff
@@ -58,6 +58,9 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 	target_link_libraries(platform_folders PRIVATE "${_CoreServices_FRAMEWORK}")
 endif()
 
+# Windows 7 or newer
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_WIN32_WINNT=0x0601 -DWINVER=0x0601")
```

Comment:
I have no idea if these are actually needed, but the modern way to do this is with `target_compile_features`

I didn't test this, but here's an example:
```cmake
if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
	# You can pass these when calling Cmake, so don't override if the user does
	if(NOT _WIN32_WINNT AND NOT WINVER)
		target_compile_features(platform_folders PRIVATE
			-D_WIN32_WINNT=0x0601
			-DWINVER=0x0601
		)
	endif()
endif()

```

Reply:
I need them to compile with my default Windows cross-compile environment (https://hub.docker.com/r/sago007/mxe_basic/).

It gives an error like this:
```
== Using MXE wrapper: /usr/lib/mxe/usr/bin/i686-w64-mingw32.static-cmake
== Using MXE toolchain: /usr/lib/mxe/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.cmake
== Using MXE runresult: /usr/lib/mxe/usr/share/cmake/modules/TryRunResults.cmake
loading initial cache file /usr/lib/mxe/usr/share/cmake/modules/TryRunResults.cmake
CMake Error at CMakeLists.txt:64 (target_compile_features):
  target_compile_features specified unknown feature "-D_WIN32_WINNT=0x0601"
  for target "platform_folders".


-- Configuring incomplete, errors occurred!
See also "/running_dir/winbuild/CMakeFiles/CMakeOutput.log".
```

Reply:
Oh, my bad. It seems you use `target_compile_definitions` and add everything but the dash, like so...

```cmake
target_compile_definitions(platform_folders PRIVATE
  _WIN32_WINNT=0x0601
  WINVER=0x0601
)
```

I think that should work? I'm just curious why it's needed, I've been able to compile on Win10 just fine without it.

Reply:
I do not use Visual Studio myself. I cross compile using mingw32. It does not support SHGetKnownFolderPath unless the target Windows version is Vista or later. I believe XP is default.
It fails with:
```
/current_dir/sago/platform_folders.cpp: In function 'std::__cxx11::string GetKnownWindowsFolder(const KNOWNFOLDERID&, const char*)':
/current_dir/sago/platform_folders.cpp:79:38: error: 'KF_FLAG_CREATE' was not declared in this scope
  hr = SHGetKnownFolderPath(folderId, KF_FLAG_CREATE, NULL, &wszPath);
                                      ^
/current_dir/sago/platform_folders.cpp:79:68: error: 'SHGetKnownFolderPath' was not declared in this scope
  hr = SHGetKnownFolderPath(folderId, KF_FLAG_CREATE, NULL, &wszPath);
```